# mayhem_program::instructions::fee_recipients::sweep_fees_admin  (yours 4384 B at .text+0x14c48; original 8008 B at .text+0x10d28)
# distance 938 instructions; evidence: called-by mayhem_program::__private::__global::sweep_fees_admin#8a63
--- original
+++ yours
@@ -1,650 +1,244 @@
-ldxdw r2, [r1+24]
-jne r2, 16, +0xf0
 ldxdw r2, [r1+8]
-stxdw [r10-608], r2
-ldxdw r2, [r2+88]
-ldxdw r4, [r2]
+stxdw [r10-232], r2
+ldxdw r5, [r2+600]
+ldxdw r4, [r5+8]
+ldxdw r0, [r4]
+add64 r0, 1
+mov64 r7, 1
+jeq r0, 0, +0x48
+mov64 r7, 0
+ldxdw r3, [r1+24]
+ldxdw r2, [r1+16]
+ldxdw r1, [r5]
+stxdw [r4], r0
+jeq r7, 1, +0x1118
+ldxdw r0, [r5+16]
+ldxdw r7, [r0]
+add64 r7, 1
+mov64 r8, 1
+jeq r7, 0, +0xa0
+mov64 r8, 0
+stxdw [r0], r7
+jeq r8, 1, +0x1118
+ldxdw r7, [r5+24]
+ldxdw r8, [r5+32]
+ldxb r9, [r5+40]
+ldxb r6, [r5+41]
+ldxb r5, [r5+42]
+stxb [r10-166], r5
+stxb [r10-167], r6
+stxb [r10-168], r9
+stxdw [r10-176], r8
+stxdw [r10-184], r7
+stxdw [r10-192], r0
+stxdw [r10-200], r4
+stxdw [r10-208], r1
+ldxdw r1, [r10-232]
+ldxdw r4, [r1+160]
+ldxdw r1, [r4+8]
+ldxdw r0, [r1]
+add64 r0, 1
+mov64 r7, 1
+jeq r0, 0, +0x158
+mov64 r7, 0
+ldxdw r5, [r4]
+stxdw [r1], r0
+jeq r7, 1, +0x1118
+ldxdw r0, [r4+16]
+ldxdw r7, [r0]
+add64 r7, 1
+mov64 r8, 1
+jeq r7, 0, +0x1a0
+mov64 r8, 0
+stxdw [r0], r7
+jeq r8, 1, +0x1118
+ldxdw r6, [r4+24]
+ldxdw r7, [r4+32]
+ldxb r8, [r4+40]
+ldxb r9, [r4+41]
+ldxb r4, [r4+42]
+stxb [r10-6], r4
+stxb [r10-7], r9
+stxb [r10-8], r8
+stxdw [r10-16], r7
+stxdw [r10-24], r6
+stxdw [r10-32], r0
+stxdw [r10-40], r1
+stxdw [r10-48], r5
+ldxdw r1, [r10-232]
+ldxdw r5, [r1+616]
+ldxdw r4, [r5+8]
+ldxdw r0, [r4]
+add64 r0, 1
+mov64 r7, 1
+jeq r0, 0, +0x258
+mov64 r7, 0
+ldxdw r1, [r5]
+stxdw [r4], r0
+jeq r7, 1, +0x1118
+ldxdw r0, [r5+16]
+ldxdw r7, [r0]
+add64 r7, 1
+mov64 r8, 1
+jeq r7, 0, +0x2a0
+mov64 r8, 0
+stxdw [r0], r7
+jeq r8, 1, +0x1118
+mov64 r6, r10
+add64 r6, -192
+stxdw [r10-240], r6
+mov64 r6, r10
+add64 r6, -200
+stxdw [r10-248], r6
+mov64 r6, r10
+add64 r6, -32
+stxdw [r10-256], r6
+mov64 r6, r10
+add64 r6, -40
+stxdw [r10-264], r6
+ldxdw r6, [r5+24]
+ldxdw r8, [r5+32]
+ldxb r9, [r5+40]
+ldxb r7, [r5+41]
+ldxb r5, [r5+42]
+stxb [r10-86], r5
+stxb [r10-87], r7
+stxb [r10-88], r9
+stxdw [r10-96], r8
+stxdw [r10-104], r6
+stxdw [r10-112], r0
+stxdw [r10-120], r4
+stxdw [r10-128], r1
+ldxdw r4, [r10-232]
+ldxdw r4, [r4+88]
+ldxdw r4, [r4]
 ldxdw r5, [r4+24]
-mov64 r7, 0
-mov64 r3, 1
-lddw r0, 0x100000000f0a098
-mov64 r2, 1
-jeq r5, r0, +0x268
-mov64 r2, 0
-ldxdw r0, [r4+16]
-lddw r6, 0x553beb1adc39c4da
-mov64 r5, 1
-jne r0, r6, +0x290
-ldxdw r0, [r4]
-lddw r6, 0x8481abfe57889b06
-mov64 r8, 1
-jeq r0, r6, +0x2c0
-mov64 r8, 0
-ldxdw r4, [r4+8]
-lddw r0, 0x35c01846637f68fb
-jne r4, r0, +0x2e0
-ja +0x2e8
+stxdw [r10-136], r5
+ldxdw r5, [r4+16]
+stxdw [r10-144], r5
+ldxdw r5, [r4+8]
+stxdw [r10-152], r5
+ldxdw r4, [r4]
+stxdw [r10-160], r4
+ldxdw r4, [r10-200]
+mov64 r5, r10
+add64 r5, -160
+stxdw [r10-4080], r5
+stxdw [r10-4088], r1
 mov64 r1, r10
 add64 r1, -48
-lddw r2, .rodata[10000000]
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#fdf6
-stdw [r10-360], 0
-stdw [r10-368], 1
-stdw [r10-376], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#91b5, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#a290]
-stxdw [r10-312], r1
-mov64 r1, r10
-add64 r1, -376
-stxdw [r10-320], r1
-lddw r1, 0xe0000020
-stxdw [r10-304], r1
-mov64 r2, r10
-add64 r2, -320
-lddw r1, .rodata[10000000]
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#0036
-jne r0, 0, +0x1e80
-ldxdw r1, [r10-360]
-stxdw [r10-192], r1
-ldxdw r1, [r10-368]
-stxdw [r10-200], r1
-ldxdw r1, [r10-376]
-stxdw [r10-208], r1
-ldxdw r1, [r10-48]
-stxdw [r10-232], r1
+stxdw [r10-4096], r1
+stdw [r10-4064], 1
+stdw [r10-4072], -1
+mov64 r1, r10
+add64 r1, -224
+mov64 r5, r10
+call mayhem_program::instructions::fee_recipients::sweep_remaining#7d4d
+mov64 r8, r10
+add64 r8, -112
+mov64 r1, r10
+add64 r1, -120
+ldxdw r9, [r10-224]
+jne r9, 2, +0x618
+ldxdw r9, [r10-216]
+ldxdw r2, [r10-120]
+ldxdw r3, [r2]
+add64 r3, -1
+stxdw [r2], r3
+jne r3, 0, +0x4a8
+call alloc::rc::Rc<T,A>::drop_slow#7561
+ldxdw r1, [r10-112]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x4e0
+mov64 r1, r8
+call alloc::rc::Rc<T,A>::drop_slow#d793
 ldxdw r1, [r10-40]
-stxdw [r10-224], r1
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x518
+ldxdw r1, [r10-264]
+call alloc::rc::Rc<T,A>::drop_slow#7561
 ldxdw r1, [r10-32]
-stxdw [r10-216], r1
-lddw r1, .rodata[70726f6772616d73]
-stxdw [r10-256], r1
-stw [r10-112], 6016
-stb [r10-184], 2
-stw [r10-240], 399
-stdw [r10-248], 58
-stdw [r10-264], 0
-mov64 r1, r10
-add64 r1, -264
-call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-ja +0x1480
-ldxdw r0, [r4+16]
-lddw r6, 0x553beb1adc39c4da
-mov64 r5, 1
-jeq r0, r6, +0x98
-mov64 r5, 0
-ldxdw r0, [r4]
-lddw r6, 0x8481abfe57889b06
-mov64 r8, 1
-jne r0, r6, +0xc0
-ldxdw r4, [r4+8]
-lddw r0, 0x35c01846637f68fb
-jeq r4, r0, +0x2e8
-mov64 r3, 0
-mov64 r4, r8
-and64 r4, r3
-and64 r4, r5
-and64 r4, r2
-mov64 r2, r10
-add64 r2, -248
-stxdw [r10-640], r2
-mov64 r2, r10
-add64 r2, -256
-stxdw [r10-648], r2
-mov64 r2, r10
-add64 r2, -248
-stxdw [r10-624], r2
-mov64 r2, r10
-add64 r2, -256
-stxdw [r10-632], r2
-mov64 r2, r10
-add64 r2, -44
-stxdw [r10-584], r2
-mov64 r2, r10
-add64 r2, -252
-stxdw [r10-592], r2
-mov64 r2, r10
-add64 r2, -80
-stxdw [r10-576], r2
-mov64 r2, r10
-add64 r2, -352
-stxdw [r10-544], r2
-mov64 r2, r10
-add64 r2, -368
-stxdw [r10-616], r2
-mov64 r2, r10
-add64 r2, -240
-stxdw [r10-552], r2
-ldxdw r2, [r10-608]
-ldxdw r3, [r2+616]
-stxdw [r10-568], r3
-ldxdw r8, [r2+160]
-ldxdw r2, [r2+600]
-stxdw [r10-600], r2
-ldxdw r1, [r1+16]
-stxdw [r10-528], r1
-and64 r4, 1
-stxdw [r10-560], r4
-mov64 r9, 0
-stxdw [r10-536], r8
-ja +0x4e0
-stdw [r6+16], -1
-ldxdw r1, [r6+24]
-ldxdw r2, [r6+32]
-stxdw [r10-256], r2
-stxdw [r10-264], r1
-stdw [r10-248], 0
-mov64 r2, r10
-add64 r2, -264
-ldxdw r1, [r10-616]
-call <mayhem_program::instructions::fee_recipients::FeeRecipient as anchor_lang::AccountSerialize>::try_serialize#4102
-ldxdw r2, [r6+16]
-add64 r2, 1
-stxdw [r6+16], r2
-jne r0, 2, +0x1280
-add64 r7, 96
-jeq r7, 768, +0x1440
-stxdw [r10-392], r9
-ldxdw r9, [r10-528]
-add64 r9, r7
-mov64 r1, r10
-add64 r1, -264
-mov64 r2, r9
-call anchor_lang::accounts::account::Account<T>::try_from#8f53
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x550
+ldxdw r1, [r10-256]
+call alloc::rc::Rc<T,A>::drop_slow#d793
+ldxdw r1, [r10-200]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x588
 ldxdw r1, [r10-248]
-ldxdw r0, [r10-256]
-ldxdw r2, [r10-264]
-jeq r2, 0, +0x1280
-ldxdw r5, [r10-552]
-ldxdw r3, [r5+16]
-ldxdw r4, [r10-544]
-stxdw [r4+16], r3
-ldxdw r3, [r5+8]
-stxdw [r4+8], r3
-ldxdw r3, [r5]
-stxdw [r4], r3
-stxdw [r10-360], r1
-stxdw [r10-368], r0
-stxdw [r10-376], r2
-mov64 r2, r9
-add64 r2, 48
-mov64 r1, r10
-add64 r1, -264
-call anchor_lang::accounts::interface_account::InterfaceAccount<T>::try_from_unchecked#f02f
-ldxw r1, [r10-88]
-jeq r1, 2, +0x1298
-ldxdw r6, [r10-160]
-ldxdw r5, [r10-232]
-ldxb r1, [r10-335]
-stxb [r10-321], r1
-mov64 r1, r10
-add64 r1, -265
-stxdw [r10-288], r1
-mov64 r1, r10
-add64 r1, -321
-stxdw [r10-304], r1
-lddw r1, .rodata[6665652d72656369]
-stxdw [r10-320], r1
-ldxb r1, [r10-336]
-stxb [r10-265], r1
-stdw [r10-280], 1
-stdw [r10-296], 1
-stdw [r10-312], 13
-jeq r6, 0, +0xcb8
-ldxdw r1, [r10-568]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-24], r2
-ldxdw r2, [r1+16]
-stxdw [r10-32], r2
-ldxdw r2, [r1+8]
-stxdw [r10-40], r2
-ldxdw r1, [r1]
-stxdw [r10-48], r1
-ldxdw r3, [r5+8]
-ldxdw r1, [r3]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x6e0
-mov64 r2, 0
-ldxdw r4, [r5]
-stxdw [r3], r1
-jeq r2, 1, +0x1e30
-ldxdw r0, [r5+16]
-ldxdw r1, [r0]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x728
-mov64 r2, 0
-stxdw [r0], r1
-jeq r2, 1, +0x1e30
-stxdw [r10-384], r7
-ldxdw r1, [r8+8]
-stxdw [r10-400], r1
-ldxdw r1, [r1]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x778
-mov64 r2, 0
-ldxdw r7, [r8]
-stxdw [r10-464], r7
-ldxb r7, [r5+42]
-stxdw [r10-456], r7
-ldxb r7, [r5+41]
-stxdw [r10-448], r7
-ldxb r7, [r5+40]
-stxdw [r10-440], r7
-ldxdw r7, [r5+32]
-stxdw [r10-432], r7
-ldxdw r5, [r5+24]
-stxdw [r10-424], r5
-ldxdw r5, [r10-400]
-stxdw [r5], r1
-jeq r2, 1, +0x1e30
-ldxdw r1, [r8+16]
-stxdw [r10-408], r1
-ldxdw r1, [r1]
-add64 r1, 1
-mov64 r2, 1
-jeq r1, 0, +0x828
-mov64 r2, 0
-ldxdw r5, [r10-408]
-stxdw [r5], r1
-ldxdw r1, [r10-384]
-jeq r2, 1, +0x1e30
-stxdw [r10-488], r0
-stxdw [r10-480], r4
-stxdw [r10-472], r3
-stxdw [r10-416], r6
-ldxdw r4, [r10-376]
-ldxdw r1, [r4+8]
-ldxdw r2, [r1]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x8a0
-mov64 r3, 0
-ldxb r7, [r8+42]
-ldxb r6, [r8+41]
-ldxb r0, [r8+40]
-ldxdw r5, [r8+32]
-stxdw [r10-504], r5
-ldxdw r8, [r8+24]
-stxdw [r10-496], r8
-ldxdw r8, [r4]
-stxdw [r1], r2
-jeq r3, 1, +0x1e30
-stxdw [r10-512], r1
-ldxdw r2, [r4+16]
+call alloc::rc::Rc<T,A>::drop_slow#7561
+ldxdw r1, [r10-192]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x5c0
+ldxdw r1, [r10-240]
+call alloc::rc::Rc<T,A>::drop_slow#d793
+mov64 r1, r10
+add64 r1, -128
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-128]
+jne r1, 1, +0x780
+ldxw r2, [r10-120]
+ldxw r1, [r10-124]
+call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
+mov64 r9, r0
+mov64 r6, r1
+ja +0x768
+ldxdw r6, [r10-216]
+ldxdw r2, [r10-120]
 ldxdw r3, [r2]
-add64 r3, 1
-mov64 r1, 1
-jeq r3, 0, +0x928
-mov64 r1, 0
+add64 r3, -1
 stxdw [r2], r3
-jeq r1, 1, +0x1e30
-ldxdw r1, [r4+24]
-stxdw [r10-520], r1
-ldxdw r3, [r4+32]
-ldxb r1, [r4+40]
-ldxb r5, [r4+41]
-ldxb r4, [r4+42]
-stxb [r10-102], r4
-stxb [r10-103], r5
-stxb [r10-104], r1
-stxdw [r10-112], r3
-ldxdw r1, [r10-520]
-stxdw [r10-120], r1
-stxdw [r10-128], r2
-ldxdw r1, [r10-512]
-stxdw [r10-136], r1
-stxdw [r10-144], r8
-stxb [r10-150], r7
-stxb [r10-151], r6
-stxb [r10-152], r0
-ldxdw r1, [r10-504]
-stxdw [r10-160], r1
-ldxdw r1, [r10-496]
-stxdw [r10-168], r1
-ldxdw r1, [r10-408]
-stxdw [r10-176], r1
-ldxdw r1, [r10-400]
-stxdw [r10-184], r1
-ldxdw r1, [r10-464]
-stxdw [r10-192], r1
-ldxdw r1, [r10-456]
-stxb [r10-198], r1
-ldxdw r1, [r10-448]
-stxb [r10-199], r1
-ldxdw r1, [r10-440]
-stxb [r10-200], r1
-ldxdw r1, [r10-432]
-stxdw [r10-208], r1
-ldxdw r1, [r10-424]
-stxdw [r10-216], r1
-ldxdw r1, [r10-488]
-stxdw [r10-224], r1
-ldxdw r1, [r10-472]
-stxdw [r10-232], r1
-ldxdw r1, [r10-480]
-stxdw [r10-240], r1
-mov64 r1, r10
-add64 r1, -320
-stxdw [r10-16], r1
-stdw [r10-8], 3
-stdw [r10-248], 0
-stdw [r10-256], 8
-stdw [r10-264], 0
-ldxdw r1, [r10-24]
-ldxdw r2, [r10-576]
-stxdw [r2+24], r1
+jne r3, 0, +0x650
+call alloc::rc::Rc<T,A>::drop_slow#7561
+ldxdw r1, [r10-112]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x688
+mov64 r1, r8
+call alloc::rc::Rc<T,A>::drop_slow#d793
+ldxdw r1, [r10-40]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x6c0
+ldxdw r1, [r10-264]
+call alloc::rc::Rc<T,A>::drop_slow#7561
 ldxdw r1, [r10-32]
-stxdw [r2+16], r1
-ldxdw r1, [r10-40]
-stxdw [r2+8], r1
-ldxdw r1, [r10-48]
-stxdw [r2], r1
-mov64 r1, r10
-add64 r1, -16
-stxdw [r10-96], r1
-stdw [r10-88], 1
-mov64 r1, r10
-add64 r1, -264
-ldxdw r6, [r10-416]
-mov64 r2, r6
-call anchor_spl::token::transfer#8298
-jne r0, 2, +0x1280
-mov64 r4, r6
-ldxdw r2, [r10-392]
-add64 r4, r2
-mov64 r1, 1
-jlt r4, r2, +0xba0
-mov64 r1, 0
-ldxdw r7, [r10-384]
-ldxdw r8, [r10-536]
-jeq r1, 1, +0x1e38
-ldxdw r1, [r10-560]
-jeq r1, 0, +0xcd0
-stxdw [r10-392], r4
-ldxdw r1, [r10-376]
-call solana_account_info::AccountInfo::lamports#aa02
-mov64 r8, r0
-mov64 r1, r10
-add64 r1, -264
-call solana_sysvar::rent::<impl solana_sysvar::Sysvar for solana_rent::Rent>::get#3513
-ldxw r1, [r10-264]
-jeq r1, 1, +0x12b8
-ldxdw r4, [r10-592]
-ldxw r1, [r4+16]
-ldxw r2, [r10-256]
-ldxdw r3, [r10-584]
-stxw [r3+16], r1
-ldxdw r1, [r4+8]
-stxdw [r3+8], r1
-ldxdw r1, [r4]
-stxdw [r3], r1
-stxw [r10-48], r2
-mov64 r1, r9
-call solana_account_info::AccountInfo::data_len#5963
-mov64 r1, r10
-add64 r1, -48
-mov64 r2, r0
-call solana_rent::Rent::minimum_balance#0f61
-jlt r8, r0, +0x1ee8
-jne r8, r0, +0xd48
-mov64 r8, r6
-ldxdw r9, [r10-392]
-ja +0x11d8
-ldxdw r4, [r10-392]
-ldxdw r1, [r10-560]
-jne r1, 0, +0xbc8
-ldxdw r3, [r10-352]
-mov64 r1, r3
-add64 r1, r6
-mov64 r2, 1
-jlt r1, r3, +0xd00
-mov64 r2, 0
-and64 r2, 1
-jne r2, 0, +0x1ed0
-stxdw [r10-352], r1
-mov64 r9, r4
-ldxdw r6, [r10-376]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x6f8
+ldxdw r1, [r10-256]
+call alloc::rc::Rc<T,A>::drop_slow#d793
+ldxdw r1, [r10-200]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x730
+ldxdw r1, [r10-248]
+call alloc::rc::Rc<T,A>::drop_slow#7561
+ldxdw r1, [r10-192]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x768
+ldxdw r1, [r10-240]
+call alloc::rc::Rc<T,A>::drop_slow#d793
+mov64 r0, r9
 mov64 r1, r6
-call anchor_lang::common::is_closed#7dce
-jne r0, 0, +0x4d0
-ja +0x1248
-ldxdw r2, [r10-376]
-ldxdw r1, [r2+8]
-ldxdw r4, [r1]
-add64 r4, 1
-mov64 r5, 1
-jeq r4, 0, +0xd80
-mov64 r5, 0
-ldxdw r3, [r2]
-stxdw [r1], r4
-jeq r5, 1, +0x1e30
-stxdw [r10-384], r7
-ldxdw r4, [r2+16]
-ldxdw r5, [r4]
-add64 r5, 1
-mov64 r7, 1
-jeq r5, 0, +0xdd0
-mov64 r7, 0
-stxdw [r4], r5
-jeq r7, 1, +0x1e30
-stxdw [r10-416], r6
-ldxdw r5, [r2+24]
-ldxdw r7, [r2+32]
-ldxb r9, [r2+40]
-ldxb r6, [r2+41]
-ldxb r2, [r2+42]
-stxb [r10-222], r2
-stxb [r10-223], r6
-stxb [r10-224], r9
-stxdw [r10-232], r7
-stxdw [r10-240], r5
-stxdw [r10-248], r4
-stxdw [r10-264], r3
-stxdw [r10-256], r1
-ldxdw r2, [r1+16]
-jne r2, 0, +0x12e8
-sub64 r8, r0
-stdw [r1+16], -1
-ldxdw r2, [r1+24]
-ldxdw r3, [r2]
-jlt r3, r8, +0x1f18
-sub64 r3, r8
-stxdw [r2], r3
-ldxdw r2, [r1+16]
-add64 r2, 1
-stxdw [r1+16], r2
-ldxdw r1, [r10-256]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-ldxdw r7, [r10-600]
-jne r2, 0, +0xef0
-ldxdw r1, [r10-632]
-call alloc::rc::Rc<T,A>::drop_slow#7561
-ldxdw r1, [r10-248]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-jne r2, 0, +0xf28
-ldxdw r1, [r10-624]
-call alloc::rc::Rc<T,A>::drop_slow#d793
-ldxdw r1, [r7+8]
-ldxdw r3, [r1]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xf58
-mov64 r4, 0
-ldxdw r2, [r7]
-stxdw [r1], r3
-jeq r4, 1, +0x1e30
-ldxdw r3, [r7+16]
-ldxdw r4, [r3]
-add64 r4, 1
-mov64 r5, 1
-jeq r4, 0, +0xfa0
-mov64 r5, 0
-stxdw [r3], r4
-jeq r5, 1, +0x1e30
-ldxdw r4, [r7+24]
-ldxdw r5, [r7+32]
-ldxb r0, [r7+40]
-ldxb r6, [r7+41]
-ldxb r7, [r7+42]
-stxb [r10-222], r7
-stxb [r10-223], r6
-stxb [r10-224], r0
-stxdw [r10-232], r5
-stxdw [r10-240], r4
-stxdw [r10-248], r3
-stxdw [r10-264], r2
-stxdw [r10-256], r1
-ldxdw r2, [r1+16]
-jne r2, 0, +0x1388
-stdw [r1+16], -1
-ldxdw r2, [r1+24]
-ldxdw r5, [r2]
-mov64 r3, r5
-add64 r3, r8
-mov64 r4, 1
-jlt r3, r5, +0x1068
-mov64 r4, 0
-and64 r4, 1
-ldxdw r7, [r10-384]
-ldxdw r6, [r10-416]
-jne r4, 0, +0x1f30
-stxdw [r2], r3
-ldxdw r2, [r1+16]
-add64 r2, 1
-stxdw [r1+16], r2
-ldxdw r1, [r10-256]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-jne r2, 0, +0x10e0
-ldxdw r1, [r10-648]
-call alloc::rc::Rc<T,A>::drop_slow#7561
-ldxdw r1, [r10-248]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-jeq r2, 0, +0x1150
-mov64 r2, r6
-add64 r2, r8
-mov64 r1, 1
-jlt r2, r6, +0x1180
-mov64 r1, 0
-and64 r1, 1
-ldxdw r2, [r10-392]
-jeq r1, 0, +0x1198
-ja +0x1e50
-ldxdw r1, [r10-640]
-call alloc::rc::Rc<T,A>::drop_slow#d793
-mov64 r2, r6
-add64 r2, r8
-mov64 r1, 1
-jge r2, r6, +0x1128
-and64 r1, 1
-ldxdw r2, [r10-392]
-jne r1, 0, +0x1e50
-mov64 r9, r8
-add64 r9, r2
-mov64 r1, 1
-jlt r9, r2, +0x11c0
-mov64 r1, 0
-and64 r1, 1
-jne r1, 0, +0x1e68
-add64 r8, r6
-ldxdw r3, [r10-360]
-mov64 r1, r3
-add64 r1, r8
-mov64 r2, 1
-jlt r1, r3, +0x1208
-mov64 r2, 0
-and64 r2, 1
-ldxdw r8, [r10-536]
-jne r2, 0, +0x1f00
-stxdw [r10-360], r1
-ldxdw r6, [r10-376]
-mov64 r1, r6
-call anchor_lang::common::is_closed#7dce
-jne r0, 0, +0x4d0
-ldxdw r6, [r6+16]
-ldxdw r1, [r6+16]
-jeq r1, 0, +0x460
-mov64 r1, 11
-mov64 r2, 0
-call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-jeq r0, 2, +0x4d0
-mov64 r9, r1
-jeq r0, 2, +0x1440
-ja +0x1488
-ldxdw r0, [r10-264]
-ldxdw r9, [r10-256]
-jeq r0, 2, +0x1440
-ja +0x1488
-ldxw r2, [r10-256]
-ldxw r1, [r10-260]
-call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-mov64 r9, r1
-jeq r0, 2, +0x1440
-ja +0x1488
-mov64 r1, 11
-mov64 r2, 0
-call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-mov64 r9, r1
-ldxdw r1, [r10-256]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-jne r2, 0, +0x1350
-ldxdw r1, [r10-632]
-mov64 r6, r0
-call alloc::rc::Rc<T,A>::drop_slow#7561
-mov64 r0, r6
-ldxdw r1, [r10-248]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-jne r2, 0, +0x12a8
-ldxdw r1, [r10-624]
-ja +0x1420
-mov64 r1, 11
-mov64 r2, 0
-call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-mov64 r9, r1
-ldxdw r1, [r10-256]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-jne r2, 0, +0x13f0
-ldxdw r1, [r10-648]
-mov64 r6, r0
-call alloc::rc::Rc<T,A>::drop_slow#7561
-mov64 r0, r6
-ldxdw r1, [r10-248]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-jne r2, 0, +0x12a8
-ldxdw r1, [r10-640]
-mov64 r6, r0
-call alloc::rc::Rc<T,A>::drop_slow#d793
-mov64 r0, r6
-jne r0, 2, +0x1488
-mov64 r1, r10
-add64 r1, -264
-call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-264]
-jne r1, 1, +0x1498
-ldxw r2, [r10-256]
-ldxw r1, [r10-260]
-call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-mov64 r9, r1
-mov64 r1, r9
 exit
-ldxdw r6, [r10-224]
+ldxdw r8, [r10-88]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r1, [r1]
@@ -652,63 +246,64 @@
 add64 r3, -256
 mov64 r2, 0
 mov64 r4, 1
-jgt r3, r1, +0x1538
+jle r3, r1, +0x810
+jeq r4, 0, +0x820
+lddw r3, 0x300007f00
+jne r1, 0, +0x840
+lddw r1, 0x300000007
+jgt r3, r1, +0x860
+ja +0xb30
 mov64 r4, 0
-jeq r4, 0, +0x1540
+jne r4, 0, +0x7d8
+mov64 r2, r3
 lddw r3, 0x300007f00
-jeq r1, 0, +0x1560
+jeq r1, 0, +0x7f0
 mov64 r3, r2
 lddw r1, 0x300000007
-jgt r3, r1, +0x1578
-ja +0x1778
-jne r4, 0, +0x14f8
-mov64 r2, r3
-lddw r3, 0x300007f00
-jne r1, 0, +0x1510
-lddw r1, 0x300000007
-jle r3, r1, +0x1778
+jle r3, r1, +0xb30
 lddw r1, 0x300000000
 stxdw [r1], r3
-stxdw [r3+16], r6
+stxdw [r3+16], r8
 stxdw [r3+8], r9
 lddw r1, 0x98956d65fd8682da
 stxdw [r3], r1
-stxdw [r10-264], r3
-stdw [r10-256], 24
-mov64 r1, r10
-add64 r1, -264
+stxdw [r10-128], r3
+stdw [r10-120], 24
+mov64 r1, r10
+add64 r1, -128
 mov64 r7, 1
 mov64 r2, 1
 call anchor_lang::solana_program::log::sol_log_data#9398
-ldxdw r1, [r10-608]
+ldxdw r1, [r10-232]
 ldxdw r1, [r1+632]
-ldxdw r8, [r1+8]
-ldxdw r2, [r8]
+ldxdw r4, [r1+8]
+ldxdw r2, [r4]
 add64 r2, 1
-jeq r2, 0, +0x1628
+jeq r2, 0, +0x910
 mov64 r7, 0
-ldxdw r3, [r1]
-stxdw [r10-384], r3
-stxdw [r8], r2
-jeq r7, 1, +0x1e30
-ldxdw r7, [r1+16]
-ldxdw r2, [r7]
+ldxdw r6, [r1]
+stxdw [r4], r2
+jeq r7, 1, +0x1118
+ldxdw r5, [r1+16]
+ldxdw r2, [r5]
 add64 r2, 1
 mov64 r3, 1
-jeq r2, 0, +0x1678
+jeq r2, 0, +0x958
 mov64 r3, 0
-stxdw [r7], r2
-jeq r3, 1, +0x1e30
+stxdw [r5], r2
+jeq r3, 1, +0x1118
+stxdw [r10-248], r5
+stxdw [r10-232], r4
 ldxb r2, [r1+42]
-stxdw [r10-440], r2
+stxdw [r10-288], r2
 ldxb r2, [r1+41]
-stxdw [r10-432], r2
+stxdw [r10-280], r2
 ldxb r2, [r1+40]
-stxdw [r10-424], r2
+stxdw [r10-272], r2
 ldxdw r2, [r1+32]
-stxdw [r10-416], r2
+stxdw [r10-264], r2
 ldxdw r1, [r1+24]
-stxdw [r10-408], r1
+stxdw [r10-256], r1
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r1, [r1]
@@ -716,60 +311,61 @@
 add64 r3, -256
 mov64 r2, 0
 mov64 r4, 1
-jgt r3, r1, +0x17a0
-mov64 r4, 0
-stxdw [r10-400], r7
-jeq r4, 0, +0x17b0
+jle r3, r1, +0xad0
+jeq r4, 0, +0xae0
 lddw r7, 0x300007f00
-jeq r1, 0, +0x17d0
-mov64 r7, r2
-stxdw [r10-392], r8
+jne r1, 0, +0xb00
+mov64 r2, r9
+stxdw [r10-240], r6
 lddw r1, 0x300000007
-jgt r7, r1, +0x17f0
-mov64 r1, 1
-mov64 r2, 256
-lddw r3, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:13:1)
-call alloc::raw_vec::handle_error#4993
-stxdw [r10-400], r7
-jne r4, 0, +0x1738
-mov64 r2, r3
-lddw r7, 0x300007f00
-jne r1, 0, +0x1750
-stxdw [r10-392], r8
-lddw r1, 0x300000007
-jle r7, r1, +0x1778
-lddw r8, 0x300000000
-stxdw [r8], r7
-stxdw [r7+16], r6
-stxdw [r7+8], r9
+jle r7, r1, +0xb30
+lddw r9, 0x300000000
+stxdw [r9], r7
+stxdw [r7+16], r8
+stxdw [r7+8], r2
 lddw r1, 0x98956d65fd8682da
 stxdw [r7], r1
 mov64 r1, 0
 mov64 r6, 32
-jsge r6, 0, +0x1868
+jsge r6, 0, +0xb58
 mov64 r2, 32
 lddw r3, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/core/src/iter/traits/iterator.rs:2027:9)
 call alloc::raw_vec::handle_error#4993
+mov64 r4, 0
+jne r4, 0, +0xa18
+mov64 r2, r3
+lddw r7, 0x300007f00
+jeq r1, 0, +0xa30
+mov64 r7, r2
+mov64 r2, r9
+stxdw [r10-240], r6
+lddw r1, 0x300000007
+jgt r7, r1, +0xa58
+mov64 r1, 1
+mov64 r2, 256
+lddw r3, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:33:1)
+call alloc::raw_vec::handle_error#4993
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
-ldxdw r1, [r8]
+ldxdw r1, [r9]
 lddw r3, 0x300008000
-jeq r1, 0, +0x1898
+jeq r1, 0, +0xb88
 mov64 r3, r1
 mov64 r2, r3
 sub64 r2, r6
 mov64 r6, 0
 mov64 r1, 1
 mov64 r4, 1
-jgt r2, r3, +0x18f8
+jle r2, r3, +0xbe0
+jeq r4, 0, +0xbf0
+lddw r2, 0x300000007
+jgt r6, r2, +0xc10
+ja +0xab0
 mov64 r4, 0
-jeq r4, 0, +0x1900
-lddw r2, 0x300000007
-jgt r6, r2, +0x1920
-ja +0x1848
-jne r4, 0, +0x18d8
+jne r4, 0, +0xbc0
 mov64 r6, r2
 lddw r2, 0x300000007
-jle r6, r2, +0x1848
+jle r6, r2, +0xab0
+lddw r8, 0x300000000
 stxdw [r8], r6
 lddw r1, 0x1d9acb512ea545e4
 stxdw [r6], r1
@@ -827,66 +423,54 @@
 mov64 r4, 1
 mov64 r1, r2
 add64 r1, -34
-jgt r1, r2, +0x1b60
-mov64 r4, 0
-ldxdw r8, [r10-392]
-jeq r4, 0, +0x1b70
+jle r1, r2, +0xff8
+jeq r4, 0, +0x1008
 lddw r1, 0x300007fde
-jeq r2, 0, +0x1b90
-mov64 r1, r3
+jne r2, 0, +0x1028
 lddw r2, 0x300000008
-jge r1, r2, +0x1ba8
-mov64 r1, 1
-mov64 r2, 34
-call alloc::alloc::handle_alloc_error#decd
-ldxdw r8, [r10-392]
-jne r4, 0, +0x1b10
-mov64 r3, r1
-lddw r1, 0x300007fde
-jne r2, 0, +0x1b28
-lddw r2, 0x300000008
-jlt r1, r2, +0x1b48
+jlt r1, r2, +0x1048
 lddw r2, 0x300000000
 stxdw [r2], r1
-ldxdw r9, [r10-384]
-ldxdw r2, [r9]
-ldxdw r3, [r9+8]
-ldxdw r4, [r9+16]
-ldxdw r5, [r9+24]
+ldxdw r7, [r10-240]
+ldxdw r2, [r7]
+ldxdw r3, [r7+8]
+ldxdw r4, [r7+16]
+ldxdw r5, [r7+24]
 stxdw [r1+24], r5
 stxdw [r1+16], r4
 stxdw [r1+8], r3
 stxdw [r1], r2
 sth [r1+32], 1
-stxdw [r10-312], r1
-stdw [r10-304], 1
-stdw [r10-320], 1
-mov64 r7, r10
-add64 r7, -264
+stxdw [r10-40], r1
+stdw [r10-32], 1
+stdw [r10-48], 1
+mov64 r9, r10
+add64 r9, -128
 mov64 r5, r10
-add64 r5, -320
-mov64 r1, r7
+add64 r5, -48
+mov64 r1, r9
 lddw r2, .rodata[052ae5d7a7daa724a6eab0a7295491855ad4a0671660674c4e034559803d65a3]
 mov64 r3, r6
 mov64 r4, 32
 call solana_instruction::Instruction::new_with_bytes#33e2
-ldxdw r1, [r10-440]
-stxb [r10-278], r1
-ldxdw r1, [r10-432]
-stxb [r10-279], r1
-ldxdw r1, [r10-424]
-stxb [r10-280], r1
-ldxdw r1, [r10-416]
-stxdw [r10-288], r1
-ldxdw r1, [r10-408]
-stxdw [r10-296], r1
-ldxdw r1, [r10-400]
-stxdw [r10-304], r1
-stxdw [r10-312], r8
-stxdw [r10-320], r9
+ldxdw r1, [r10-288]
+stxb [r10-6], r1
+ldxdw r1, [r10-280]
+stxb [r10-7], r1
+ldxdw r1, [r10-272]
+stxb [r10-8], r1
+ldxdw r1, [r10-264]
+stxdw [r10-16], r1
+ldxdw r1, [r10-256]
+stxdw [r10-24], r1
+ldxdw r1, [r10-248]
+stxdw [r10-32], r1
+ldxdw r1, [r10-232]
+stxdw [r10-40], r1
+stxdw [r10-48], r7
 mov64 r2, r10
-add64 r2, -320
-mov64 r1, r7
+add64 r2, -48
+mov64 r1, r9
 mov64 r3, 1
 lddw r4, .data.rel.ro+obj
 mov64 r5, 1
@@ -895,56 +479,42 @@
 mov64 r1, r0
 lsh64 r1, 32
 rsh64 r1, 32
-jne r1, 26, +0x1d60
-mov64 r0, 2
-ja +0x1d80
+jne r1, 26, +0x1060
+mov64 r9, 2
+ldxdw r2, [r10-232]
+ja +0x1088
+mov64 r4, 0
+jne r4, 0, +0xe00
+mov64 r3, r1
+lddw r1, 0x300007fde
+jeq r2, 0, +0xe18
+mov64 r1, r3
+lddw r2, 0x300000008
+jge r1, r2, +0xe30
+mov64 r1, 1
+mov64 r2, 34
+call alloc::alloc::handle_alloc_error#decd
 mov64 r1, r0
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-mov64 r9, r1
-ldxdw r8, [r10-312]
-ldxdw r1, [r8]
+mov64 r9, r0
+mov64 r6, r1
+ldxdw r2, [r10-40]
+ldxdw r1, [r2]
 add64 r1, -1
-stxdw [r8], r1
-jne r1, 0, +0x1dc8
-mov64 r1, r10
-add64 r1, -312
-mov64 r6, r0
-call alloc::rc::Rc<T,A>::drop_slow#7561
-mov64 r0, r6
-ldxdw r1, [r10-304]
-ldxdw r2, [r1]
-add64 r2, -1
-stxdw [r1], r2
-jne r2, 0, +0x1e18
-mov64 r1, r10
-add64 r1, -304
-mov64 r6, r0
-call alloc::rc::Rc<T,A>::drop_slow#d793
-mov64 r0, r6
-jne r0, 2, +0x1488
-mov64 r0, 2
-ja +0x1488
+stxdw [r2], r1
+jne r1, 0, +0x10c0
+mov64 r1, r10
+add64 r1, -40
+call alloc::rc::Rc<T,A>::drop_slow#7561
+ldxdw r1, [r10-32]
+ldxdw r2, [r1]
+add64 r2, -1
+stxdw [r1], r2
+jne r2, 0, +0x1100
+mov64 r1, r10
+add64 r1, -32
+call alloc::rc::Rc<T,A>::drop_slow#d793
+jne r9, 2, +0x768
+mov64 r9, 2
+ja +0x768
 call abort
-lddw r1, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:353:84)
-call core::option::unwrap_failed#2d9b
-lddw r1, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:367:22)
-call core::option::unwrap_failed#2d9b
-lddw r1, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:368:86)
-call core::option::unwrap_failed#2d9b
-mov64 r3, r10
-add64 r3, -264
-lddw r1, str("a Display implementation returned an error unexpectedly")
-mov64 r2, 55
-lddw r4, table[3:<core::fmt::Error as core::fmt::Debug>::fmt#aaa5]
-lddw r5, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/string.rs:2830:14)
-call core::result::unwrap_failed#5a05
-lddw r1, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:378:17)
-call core::panicking::panic_const::panic_const_add_overflow#3493
-lddw r1, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:358:37)
-call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:375:17)
-call core::panicking::panic_const::panic_const_add_overflow#3493
-lddw r1, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:362:17)
-call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/fee_recipients.rs:363:17)
-call core::panicking::panic_const::panic_const_add_overflow#3493
